### ISSUE TITLE
“Hide completed categories” should also hide empty categories

### ISSUE THREAD

**Issue by tflo** [Ny0n/Nys_ToDoList#14]
I like the new “Hide completed categories” feature for my Weekly and Daily tabs.

But, if this is enabled, then also _empty_ categories should be hidden.

Why?

My “categories” are actually my chars, and not all of them have always a daily or weekly task. As I have many chars, also the ones that do have tasks are usually collapsed, because of the scrolling bug[^1]. The big advantage of “Hide completed categories” is _– in theory –_ that I don’t have to expand a category (a char) to see if all tasks are completed, because the category simply is no longer visible if all is completed.

However, the _empty_ categories (chars without tasks) are still visible, and thus indistinguishable from collapsed categories that still contain open (non-favorite) tasks. As a result, I still click the collapsed empty chars, because I think there could be tasks inside.

If they were hidden (like a category with all tasks completed) then we would have unambiguity: Collapsed category visible = something not yet done is inside.

I hope it is understandable what I mean ;)

– Tom

[^1]: If the list has a certain length it is often impossible to scroll to the bottom. The scroll bar is at the bottom stop but the text is cut off at the bottom (can be a half line or one line or more). Collapsing some of the other categories usually brings the text back in the visible/scrollable area.

**Comment by Ny0n** [comment]
Hey Tom! This feature was something we had in older versions of the addon, it was simply used to clean up the list so that we only saw what was left to do, but it got lost in 6.0, so I added it back.

But now I realise that since we can have empty categories, only hiding the completed ones might not be enough. I didn't think someone would voluntarily leave empty categories, but it makes sense in your case.

The problem with hiding empty categories was simply 'how are we supposed to add new items in it?'. What I'm thinking is that I'll probably add the possibility to add items while in edit mode, that way it'll solve those sorts of problems.

About the 'scrolling bug', it's probably just me that didn't give the right amount of space for empty categories/other things, I'll look into it.

Thanks a lot for the feedback!
